### PR TITLE
Passcoderebase stable

### PR DESCRIPTION
**Pull request by huminzhi**

### CODE REVIEW

**Comment by bhariharan** [comment]
Looks good to me. 👍

**Comment by huminzhi** [comment]
@bhariharan sorry that I missed some new files, will you please help review..Thank you!
cc @khawkins

File: shared/test/test/PasscodeUITests.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        
+        passcodePage.enterPasscode(passcode)
+        backspace(1) //backspace
+        XCTAssertTrue(passcodePage.isPresented() && passcodePage.getStatus()==PasscodeStatus.Verifying)
+        passcodePage.enterPasscode(passcode.substringFromIndex(passcode.endIndex.advancedBy(-1)))
+        passcodePage.done()
+        XCTAssertFalse(passcodePage.isPresented()) //should not present passcode
+        
+        //TODO: verify background/foreground after timeout
+//        XCUIDevice().pressButton(XCUIDeviceButton.Home)
```
Comment (bhariharan) [inline shared/test/test/PasscodeUITests.swift]: Should this be commented out?

**Comment by bhariharan** [comment]
Looks good to me otherwise.

**Comment by khawkins** [comment]
Seems fine to me.

File: shared/test/test/PasscodeUITests.swift (hunk 1 above)
Comment (huminzhi) [inline shared/test/test/PasscodeUITests.swift]: ok, I should probably remove this part for now, I am hitting some issue when run app.resolve() which supposed to bring the app to foreground..

**Pull request merged**